### ISSUE TITLE
extract-bc -b not working. llvm-link: No such file or directory

### ISSUE THREAD

**Issue by nakasan617** [travitch/whole-program-llvm#119]
I am trying to extract bit-code from the executable ./hello but not available. (./hello is made of most basic helloworld C code, but I have inserted libc.a that is made in [musllvm](https://github.com/SRI-CSL/musllvm)).
The error message says 
`llvm-link: No such file or directory`
but before it was saying 
```
extract-bc -b hello -l llvmLinker
ERROR:WLLVM Failed to execute: ['llvmLinker',
 '-o=/home/yuta/yuta_experiment/experiment_sets/helloworld/hello.bc',
 '/home/yuta/yuta_experiment/experiment_sets/helloworld/.helloworld.o.bc',
 '/home/vagrant/musllvm/obj/src/env/.__libc_start_main.o.bc',
 '/home/vagrant/musllvm/obj/src/exit/.exit.o.bc',
 '/home/vagrant/musllvm/obj/src/internal/.defsysinfo.o.bc',
 '/home/vagrant/musllvm/obj/src/internal/.libc.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.printf.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.stdout.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.vfprintf.o.bc',
 '/home/vagrant/musllvm/obj/src/string/.memset.o.bc',
 '/home/vagrant/musllvm/obj/src/string/.strnlen.o.bc',
 '/home/vagrant/musllvm/obj/src/env/.__environ.o.bc',
 '/home/vagrant/musllvm/obj/src/env/.__init_tls.o.bc',
 '/home/vagrant/musllvm/obj/src/errno/.__errno_location.o.bc',
 '/home/vagrant/musllvm/obj/src/errno/.strerror.o.bc',
 '/home/vagrant/musllvm/obj/src/exit/._Exit.o.bc',
 '/home/vagrant/musllvm/obj/src/locale/.__lctrans.o.bc',
 '/home/vagrant/musllvm/obj/src/math/.__fpclassifyl.o.bc',
 '/home/vagrant/musllvm/obj/src/math/.__signbitl.o.bc',
 '/home/vagrant/musllvm/obj/src/math/.frexpl.o.bc',
 '/home/vagrant/musllvm/obj/src/multibyte/.wctomb.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.__lockfile.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.__stdio_close.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.__stdio_seek.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.__stdout_write.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.__towrite.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.fwrite.o.bc',
 '/home/vagrant/musllvm/obj/src/string/.memchr.o.bc',
 '/home/vagrant/musllvm/obj/src/string/.memcpy.o.bc',
 '/home/vagrant/musllvm/obj/src/thread/.default_attr.o.bc',
 '/home/vagrant/musllvm/obj/src/internal/.syscall_ret.o.bc',
 '/home/vagrant/musllvm/obj/src/multibyte/.wcrtomb.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.__stdio_exit.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.__stdio_write.o.bc',
 '/home/vagrant/musllvm/obj/src/stdio/.ofl.o.bc',
 '/home/vagrant/musllvm/obj/src/thread/.__lock.o.bc']
ERROR:Your llvm-link does not seem to be easy to find.
Either install it or use the -l llvmLinker option.
clam-pp --crab-devirt --devirt-resolver=sea-dsa --sea-dsa-type-aware=false hello.bc -o hello.pp.bc
make: clam-pp: Command not found
make: *** [Makefile:8: hello.pp.bc] Error 127
```

so it seems to be the problem with the linker but I am not really sure what really is the problem.
according to the sanity checker it says 
```
wllvm version: 1.3.1
wllvm released: April 10 2021

Logging output to standard error.
Logging level not set, defaulting to WARNING.

We are using clang.

The C compiler clang is:

	clang version 10.0.0-4ubuntu1

The C++ compiler clang++ is:

	clang version 10.0.0-4ubuntu1

The bitcode linker llvm-link is:

	LLVM version 14.0.0git

The bitcode archiver llvm-ar is:

	LLVM version 14.0.0git

Not using a bitcode store.
```
But it  was warning me better not to use the extract-bc before.

I am using ubuntu 20.04, and I am trying to extract bitcode so that I can devirtualize the function calls with [clam-pp](https://github.com/SRI-CSL/clam-prov) and extract dotfiles out of them. The ultimate objective is to look into the program execution (which path it took). 
I am also using proxy to go through https, but that does not seem to be relevant at this point

**Comment by ianamason** [comment]
So it is not finding your llvm-link command, when it is doing the extracting. Is it in you path?
You can probably use the -l switch to hold its hand. Not sure why the sanity checker finds it but not the extractor.

**Comment by ianamason** [comment]
Can I close this?

**Comment by nakasan617** [comment]
sorry. yes I figured out what the problem was. I installed clang with sudo apt-get install and I installed llvm-linker with llvm-project. 
That's why the sanity checker found it but the actual extract-bc did not. So my solution was nothing special, just remove llvm-project from my machine and set LLVM_LINK_NAME and LLVM_AR_NAME to llvm-10 (because in my machine, ubuntu 20.04 is the os) installed clang-10 with sudo apt-get install.
